### PR TITLE
Move to ts

### PR DESCRIPTION
**Pull request by NeRpi**

### CODE REVIEW

File: server/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "parser": "@typescript-eslint/parser",
+  "extends": ["plugin:@typescript-eslint/recommended"],
+  "parserOptions": {
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "rules": {}
+}
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/.eslintrc]: empty line

File: server/nodemon.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "watch": ["src"],
+  "ext": ".ts,.js",
+  "ignore":[],
+  "exec": "npx ts-node ./src/index.ts"
+}
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/nodemon.json]: empty line

File: server/src/controllers/auth.controller.ts [hunk 3]
```diff
@@ -1,11 +1,14 @@
 import AuthService from "../services/auth.service.js";
+import { Request, Response, NextFunction } from "express";
 
 class AuthController {
+  _authService: AuthService;
+
   constructor() {
     this._authService = new AuthService();
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/controllers/auth.controller.ts]: you can use private modifier to restrict access

File: server/src/middleware/jwt.middleware.ts [hunk 4]
```diff
@@ -15,7 +16,7 @@ const jwtMiddleware = (req, res, next) => {
     const userData = tokenService.validateAccessToken(accessToken);
     if (!userData) return next(ApiError.unauthorized("The token is not valid"));
 
-    req.user = userData;
+    // req.user = userData;
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/middleware/jwt.middleware.ts:19]: why?

File: server/src/repositories/user.repo.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
+import ApiError from "../error/api.error.js";
+import { UserEntity } from "../db/entities/user.entity.js";
+import dbConnector from "../db/db.connector.js";
+
+export const UserRepo = dbConnector.getRepository(UserEntity).extend({
+  async createUser(
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/repositories/user.repo.ts]: it's better to wrap parameters into an object and pass only one variable
data: {
name: string,
    surname: string,
    DoB: string,
    email: string,
    password: string,
    roleId: string
}

File: server/src/services/role.service.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { RoleRepo } from "../repositories/role.repo.js";
+import ApiError from "../error/api.error.js";
+
+export default class RoleService {
+  async create(role: string) {
+    return await RoleRepo.createRole(role);
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/services/role.service.ts]: I'd set roleRepo as an private attribute. It's a common practice. Usually data convertors are used in this way

File: server/src/services/token.service.ts [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
 export default class TokenService {
-  generateTokens(payload) {
+  generateTokens(payload: object) {
     try {
-      const accessToken = jwt.sign(payload, process.env.JWT_ACCESS_SECRET, {
+      const accessToken = jwt.sign(payload, process.env.JWT_ACCESS_SECRET!, {
         expiresIn: "30m",
       });
-      const refreshToken = jwt.sign(payload, process.env.JWT_REFRESH_SECRET, {
+      const refreshToken = jwt.sign(payload, process.env.JWT_REFRESH_SECRET!, {
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/services/token.service.ts:10]: for what do you need '!'?

**Review by Daniil-Zaichuk-iTechArt — CHANGES_REQUESTED** [review]

File: server/src/services/token.service.ts (hunk 7 above)
Comment (NeRpi) [reply server/src/services/token.service.ts:10]: so I promise that the object is not undefined so that I can pass it

**Pull request merged**